docs/rackhd/workflow_progress_notification.rst

```diff
@@ -0,0 +1,149 @@
+Workflow Progress Notification
+--------------------------------
+
+RackHD workflow progress feature provides message notification mechanism to indicate status of an active workflow or task. User can get to know what has been done and what is to be done for an active workflow or task with progress messages. 
+
+Workflow Progress Events 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+RackHD will publish a workflow progress message if any of below events happens:
+
+* Workflow started or finished events
+* Task started or finished events
+* RackHD marked important milestone events for an active long-run task.
+
+  In some cases RackHD can't easily get progress information, some milestones are created to divide a task into several small sections. Progress messages will be sent if any of those milestones is achieved.
+
+* Progress timer timeout for an active long-run task.
+
+  Progress of some tasks is continuous and will be updated all the time if task is active. In this case a repeating timer with fixed interval is set. Progress message will be retrieved and sent each time the timer is timeout.
+
+Progress Message Payload
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+4 attributes are used to describe progress information:
+
+=============== ======= ==============================================================================================
+properties      Type    Description
+=============== ======= ==============================================================================================
+maximum         Integer Maximum step quantity for a workflow or a task.
+                        For tasks with continuous progress, it is 100.
```

Comment:
I think this will be more reasonable:

Even for the task with continuous progress, It's up to task to decide the value of `maximum`, for example, if a task could set:
```
maximum=1000&value=250&percent=25%
```

if only `percent` is provided, then you can directly set the maximum to 100 and the value to the percent number.